### DISCUSSION TITLE
Http request

### DISCUSSION THREAD

**Discussion by gutyerrez** [laravel/framework#33833] [General] [answered]
Hello, i trying to use Http requests to execute an put on my local server, but it say's: 

``http://localhost:8000/plataform/payment/12546487546234874 cURL error 6: Could not resolve host: 54 (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)``

My code to execute this put method:

![image](https://user-images.githubusercontent.com/36475012/89971883-d6e48f00-dc32-11ea-93d1-1a542c1eaabc.png)

What i'm do incorrectly?

**Comment by mfn** [comment, marked as answer]
Can't say what your actual problem is, but `printf` looks wrong. Seems like you want to use `sprintf` here.

**Reply by gutyerrez** [reply]
`print` and `sprintf` is the same function and my problem ins't this, my problem is because i can't execute curl's from localhost to localhost, repsentative form:

localServer -> running on port 8000 and address localhost
targetServer -> running on port 8000 and address localhost

i execute curl post to targetServer from localServer, but even all servers running this generate an error, the error is "could not connect to targetServer", but targetServer is running because the targetServer is the localServer

**Reply by mfn** [reply]
> `print` and `sprintf` is the same function

Sorry to say this again and now I realize this is the problem: `printf` does _not_ return a string which yo need for `::put()`, `sprintf` does 🤷‍♀️

**Reply by gutyerrez** [reply]
I can execute put method, but now the server is in a infinity loading waiting for response, but i have declared the response to client show the code below:

```
public function show($reference) {
    $response = Http::get('http://127.0.0.1:8000/');

    $response->status();

    return $response;
}
```

I don't understand why that's can't return an response to client

**Reply by GrahamCampbell** [reply]
Run `curl -v http://127.0.0.1:8000/` in your terminal. Show us the output.

**Reply by rodrigopedra** [reply]
How are you running your local server?

If using valet or Homestead I know it fires up several php-fpm processes, so you could handle multiple requests at once.

If running `php artisan serve`, I am guessing it will handle one request at a time as it runs a single process. (guessing here, not sure, I use the unofficial cpriego/valet-linux).

If this is the case, when your original request makes an inner request to the same app, the second request is not processed as the web server is busy handling the original one.

To test it out, you could start another `php artisan serve` on another terminal window and test sending a request to the port this new process uses.

Unrelated, out of curiosity, if you are calling an endpoint from the same project, why not executing the intended code directly (maybe by dispatching a sync job or calling a method in a class) instead of making a HTTP request which would add network latency?

Also, as @mfn pointed out and per docs, `sprintf` returns the formatted string while `print` return an integer of the formatted string size.

**Reply by gutyerrez** [reply]
> Run `curl -v http://127.0.0.1:8000/` in your terminal. Show us the output.

![image](https://user-images.githubusercontent.com/36475012/90047135-d1755c00-dca7-11ea-8b1e-4d9ae4bb9cce.png)
